rabbitmq/publisher.go

```diff
@@ -163,6 +187,36 @@ func (rp *rabbitMqPublisher) publishFinalizedToExchange(finalizedBlock data.Fina
 	}
 }
 
+func (rp *rabbitMqPublisher) publishTxsToExchange(blockTxs data.BlockTxs) {
+	if rp.cfg.TxsEventsExchange != "" {
+		txsBlockBytes, err := json.Marshal(blockTxs)
+		if err != nil {
+			log.Error("could not marshal block txs event", "err", err.Error())
+			return
+		}
+
+		err = rp.publishFanout(rp.cfg.TxsEventsExchange, txsBlockBytes)
+		if err != nil {
+			log.Error("failed to publish block txs event to rabbitMQ", "err", err.Error())
+		}
+	}
+}
+
+func (rp *rabbitMqPublisher) publishScrsToExchange(blockScrs data.BlockScrs) {
+	if rp.cfg.ScrsEventsExchange != "" {
```

Comment:
refactor:
```
if rp.cfg.ScrsEventsExchange == "" {
   return
}
// the rest
```

Reply:
refactored to check exchange names in constructor